src/store/reducers/UserSlice.ts

```diff
@@ -38,8 +38,12 @@ export const counterSlice = createSlice({
     completeTodo: (state, action: PayloadAction<string>) => {
       state.todos.map((todo) => (todo.id === action.payload ? (todo.completed = !todo.completed) : todo));
     },
+    deleteTodo: (state, action: PayloadAction<string>) => {
+      const index = state.todos.findIndex((todo) => todo.id === action.payload);
+      state.todos.splice(index, 1);
```

Comment:
why don't you want to use .filter() instead of findIndex and splice?
<img width="444" alt="Screenshot 2022-09-22 at 21 44 35" src="https://user-images.githubusercontent.com/64188262/191826794-20a5f94f-ba2d-42b2-a743-631c8ac3db7b.png">

Reply:
at first I implemented it with filter ( state.todos = state.todos.filter((todo) => todo.id !== action.payload); ) , then I remembered about another option, I decided to try it. I understand that basically in such situations used filter

Reply:
Add filter instead of findIndex and splice